pkg/storage/chunk/chunk_test.go

```diff
@@ -186,14 +252,24 @@ func testChunkBatch(t *testing.T, encoding Encoding, samples int) {
 	// Check all the samples are in there.
 	iter := chunk.NewIterator(nil)
 	for i := 0; i < samples; {
-		require.True(t, iter.Scan() == chunkenc.ValFloat)
-		batch := iter.Batch(BatchSize, chunkenc.ValFloat)
-		for j := 0; j < batch.Length; j++ {
-			require.EqualValues(t, int64((i+j)*step), batch.Timestamps[j])
-			require.EqualValues(t, float64(i+j), batch.SampleValues[j])
+		require.NotEqual(t, chunkenc.ValNone, iter.Scan())
+		var batch Batch
+		switch encoding {
+		case PrometheusXorChunk:
+			batch = iter.Batch(BatchSize, chunkenc.ValFloat)
+			for j := 0; j < batch.Length; j++ {
+				require.EqualValues(t, int64((i+j)*step), batch.Timestamps[j])
+				require.EqualValues(t, float64(i+j), batch.SampleValues[j])
```

Comment:
[rant] not the fault of this PR, but Batch is not encapsulated so batch.Timestamps/batch.SampleValues will break when we fix Batch bug+performance